airbyte-server/src/main/java/io/airbyte/server/handlers/JobHistoryHandler.java

```diff
@@ -120,6 +124,25 @@ public JobDebugInfoRead getJobDebugInfo(final JobIdRequestBody jobIdRequestBody)
     return buildJobDebugInfoRead(jobinfoRead);
   }
 
+  public Optional<JobRead> getLatestRunningSyncJob(final UUID connectionId) throws IOException {
+    final List<Job> nonTerminalSyncJobsForConnection = jobPersistence.listJobsForConnectionWithStatuses(
+        connectionId,
+        Collections.singleton(ConfigType.SYNC),
+        JobStatus.NON_TERMINAL_STATUSES);
+
+    if (!nonTerminalSyncJobsForConnection.isEmpty()) {
+
+      // jobPersistence.listJobsForConnectionWithStatuses orders by created_at desc, so index 0 is the latest.
+      // there *should* only be a single running sync job for a connection regardless so .get(0) should always be
+      // what we want.
+      final Job job = nonTerminalSyncJobsForConnection.get(0);
```

Comment:
Nit for later:  We could do something like `nonTerminalSyncJobsForConnection.stream().findFirst()` which returns an optional -- either the first element is there or the list is empty/there is nothing.